tutorials/intellijDebugger.md

```diff
@@ -4,9 +4,185 @@ title: "{{ title }}"
+{% set start_in_seconds = (start_m | int)*60 + (start_s | int) %}
+{% set end_in_seconds = (end_m | int)*60 + (end_s | int) %}
+
+Relevant video segment ({{ start_m }}:{{ start_s }} - {{ end_m }}:{{ end_s }}):
+<iframe width="560" height="315" src="https://www.youtube.com/embed/1bCgzjatcr4?si=kQCMnmebYVV1QOwu&amp;start={{ start_in_seconds }}&amp;end={{ end_in_seconds }}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+{% endmacro %}
+
 # {{ title }}
 
+It’s essential to understand key debugger features and their purpose.
+
 This video (from LaunchCode) gives a pretty good explanation of how to use the IntelliJ IDEA debugger.
 
 @[youtube](1bCgzjatcr4)
 
+## Starting A Debugging Session
+
+To utilise the debugger, we need to make sure that the code is running in debug mode.
+
+### Setting Breakpoints
+
+<img src="images/intellijDebugger/breakpoint.png" height="300dp"/>
+
+<box type="info" seamless>
+
+**Video reference:** {{ show_video("3", "09", "3", "30") }}
+Set a breakpoint by clicking on the line number in the left gutter like this!
+</box>
+
+To set a breakpoint in IntelliJ IDEA:
+
+- Click on the line number in the left gutter.
+
+Breakpoints are useful for:
+
+- Pausing execution at method entry points
+- Examining the program state before complex operations
+
+### Run Debug Mode
+
+<img src="images/intellijDebugger/debuggerMode.png" height="300dp"/>
+
+<box type="info" seamless>
+
+**Video reference:** {{ show_video("3", "53", "4", "00") }}
+Run your program in debug mode to use debugging features.
+</box>
+
+Go to `Run` > `Debug 'Main'` or click on the bug icon in the top-right corner of the IntelliJ IDEA window.
+
+## Basic Debugger Features
+
+Debugger features help you step through code systematically, inspect variable values, and understand program flow during execution. Here are the primary features you'll use:
+
+### Showing Execution Point
+
+<img src="images/intellijDebugger/showingExecutionPoint.png" height="300dp"/>
+
+<box type="info" seamless>
+
+**Video reference:** {{ show_video("4", "41", "6", "06") }}
+You can see the values of each variable up to the execution point selected.
+</box>
+
+Use the values of each variable to ascertain if the logic of your code is correct.
+
+### Stepping Over Code
+
+<img src="images/intellijDebugger/stepOverCode.png" height="300dp"/>
+
+<box type="info" seamless>
+
+**Video reference:** {{ show_video("7", "30", "7", "55") }}
+Step over the current line of code to go to the next line with this button. Alternatively, you can use `F8`.
+</box>
+
+Purpose: Executes the current line of code without stepping into method calls.
+
+Usage:
+
+- Skip over lines of code that are not critical for understanding the current flow.
+- Execute the current line of code and move on to the next.
```

Comment:
Perhaps an example can be added as well, similar to what was provided for subsequent features, to improve flow of document.